### PR TITLE
Make it possible to chose editor when create workspace from Dashboard sample in e2e GUI typescript tests 7.56.x cherry pick

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?

- It made possible to chose editor (Che Theia, VS Code or Intellij Idea) when create workspace from Dashboard sample in e2e GUI typescript tests
- It fixes wrong rules in tslint.json

### Screenshot/screencast of this PR
image

### How to test this PR?

1.a. Set up TS_SELENIUM_EDITOR to code, theia or idea
1.b. Run any test witch is choose a simple from dashboard (for example from package.json)
2. Run npm "lint" script

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers
@ScrewTSW @dmytro-ndp @musienko-maxim 
Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by dmytro-ndp — APPROVED** [review]

File: tests/e2e/TestConstants.ts [hunk 1]
```diff
@@ -282,10 +282,10 @@ export const TestConstants = {
...
+    TS_SELF_HOSTED_GIT_SERVER_URL: process.env.TS_SELF_HOSTED_GIT_SERVER_URL || '10.0.104.86:10080',
 
-     /**
-      * URL of workspace devfile test repository.
-      */
-      TS_TEST_WORKSPACE_DEVFILE_REPO: process.env.TS_TEST_WORKSPACE_DEVFILE_REPO || ''
+    /**
+     * URL of workspace devfile test repository.
+     */
+    TS_TEST_WORKSPACE_DEVFILE_REPO: process.env.TS_TEST_WORKSPACE_DEVFILE_REPO || '',
```
Comment (ScrewTSW) [inline tests/e2e/TestConstants.ts]: same here, looks like there's an extra comma at the end

File: tests/e2e/pageobjects/dashboard/CreateWorkspace.ts [hunk 2]
```diff
@@ -14,53 +14,95 @@ import { DriverHelper } from '../../utils/DriverHelper';
...
+    async waitSample(sampleName: string, timeout: number = TimeoutConstants.TS_COMMON_DASHBOARD_WAIT_TIMEOUT): Promise<void> {
         Logger.debug(`CreateWorkspace.waitSample sampleName: "${sampleName}"`);
 
         const sampleLocator: By = this.getSampleLocator(sampleName);
 
         await this.driverHelper.waitVisibility(sampleLocator, timeout);
     }
 
-    async clickOnSample(sampleName: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT) {
+    async clickOnSample(sampleName: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT): Promise<void> {
```
Comment (ScrewTSW) [inline tests/e2e/pageobjects/dashboard/CreateWorkspace.ts]: Same here, I would rename it to `clickOnSampleUsingDefaultEditor` or something similar and replace it's use with `clickOnSampleForSpecificEditor` in most of the logic.

We can use this for a specific "default editor test" later

File: tests/e2e/pageobjects/dashboard/CreateWorkspace.ts [hunk 3]
```diff
@@ -14,53 +14,95 @@ import { DriverHelper } from '../../utils/DriverHelper';
...
+        return By.xpath(`//div[text()='${sampleName}']//parent::article//span[text()[
+                contains(
+                translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'),
+                '${editor}')]
+            ]//parent::a`);
+    }
+
     private getSampleLocator(sampleName: string): By {
-        Logger.trace(`CreateWorkspace.getSampleLocator sampleName: ${sampleName}`);
+        Logger.debug(`CreateWorkspace.getSampleLocator sampleName: ${sampleName}, used default editor`);
```
Comment (ScrewTSW) [inline tests/e2e/pageobjects/dashboard/CreateWorkspace.ts]: Please move the "using default editor" log message into `clickOnSample` method instead and switch it back to trace to not clutter the logs

File: tests/e2e/tslint.json [hunk 4]
```diff
@@ -121,7 +116,8 @@
       "check-decl",
       "check-operator",
       "check-separator",
-      "check-type"
+      "check-type",
+      "check-module"
     ]
   }
 }
```
Comment (ScrewTSW) [inline tests/e2e/tslint.json]: Missing new line at the end of the file. Would be awesome to add a global linter rule for this

File: tests/e2e/TestConstants.ts (hunk 1 above)
Comment (nallikaea) [reply tests/e2e/TestConstants.ts]: thank you, deleted

File: tests/e2e/pageobjects/dashboard/CreateWorkspace.ts (hunk 2 above)
Comment (nallikaea) [reply tests/e2e/pageobjects/dashboard/CreateWorkspace.ts]: name changed

File: tests/e2e/pageobjects/dashboard/CreateWorkspace.ts (hunk 3 above)
Comment (nallikaea) [reply tests/e2e/pageobjects/dashboard/CreateWorkspace.ts]: fixed

File: tests/e2e/tslint.json (hunk 4 above)
Comment (nallikaea) [reply tests/e2e/tslint.json]: thnx, added

**Pull request merged**